src/Stratis.Bitcoin.Features.Consensus/Rules/ProvenHeaderRules/ProvenHeaderCoinstakeRule.cs

```diff
@@ -268,6 +272,34 @@ private void CheckHeaderSignatureWithCoinstakeKernel(ProvenBlockHeader header, U
             ConsensusErrors.BadBlockSignature.Throw();
         }
 
+        /// <summary>
+        /// Checks if coinstake is spent on another chain.
+        /// </summary>
+        /// <param name="header">The proven block header.</param>
+        /// <param name="context">The POS rule context.</param>
+        /// <exception cref="ConsensusException">Throws utxo not found error.</exception>
+        private void CheckIfCoinstakeIsSpentOnAnotherChain(ProvenBlockHeader header, PosRuleContext context)
+        {
+            Transaction coinstake = header.Coinstake;
+            TxIn input = coinstake.Inputs[0];
+
+            int? rewindDataIndex = this.PosParent.RewindDataIndexStore.GetAsync(input.PrevOut.Hash, (int)input.PrevOut.N).GetAwaiter().GetResult();
+            if (!rewindDataIndex.HasValue) {
+                // Not sure if we should throw this exception or another.
+                ConsensusErrors.ReadTxPrevFailed.Throw();
```

Comment:
here you also need to `SetFlagAndThrow` if we cant find a rewind maybe the utxo is not in the chain yet (if we are very much ahead of consensus